uniffi_bindgen/src/interface/conventions.rs

```diff
@@ -0,0 +1,127 @@
+use std::collections::{BTreeMap, BTreeSet};
+use uniffi_meta::Type;
+use crate::{CodeOracle, ComponentInterface, Renameable};
+use crate::interface::{FfiDefinition, Record};
+
+impl ComponentInterface {
+    pub fn apply_naming_conventions<O: CodeOracle>(&mut self, oracle: O) {
+
+        for (_, t) in &mut self.types.type_definitions {
+
+            if t.name().is_some() {
+                t.rename(oracle.var_name(&t.name().unwrap()));
+
+            }
+        }
+
+        let mut known: BTreeSet<Type> = BTreeSet::new();
+
+        for t in &mut self.types.all_known_types.iter() {
+            dbg!("type in all_known {:#?}", t.clone());
+            let mut ty = t.clone();
+            if t.name().is_some() {
+                ty.rename(oracle.external_types_name(&t.name().unwrap()));
```

Comment:
I think this should also be `class_name`.  In general, `types.all_known_types` should be handled like `types.type_definitions`